java/java-analysis-impl/src/com/siyeh/ig/abstraction/BooleanParameterInspection.java

```diff
@@ -69,6 +69,11 @@ private class BooleanParameterVisitor extends BaseInspectionVisitor {
     @Override
     public void visitMethod(@NotNull PsiMethod method) {
       super.visitMethod(method);
+      // private or protected methods are never public
+      if (method.hasModifierProperty(PsiModifier.PRIVATE) || method.hasModifierProperty(PsiModifier.PROTECTED)) {
+        return;
+      }
+      // interface methods can be public even without explicit public modifier
```

Comment:
An easier fix would be to change the next if-statement to: 
```
      if (!method.hasModifierProperty(PsiModifier.PUBLIC)) {
        return;
      }
```
Because `hasModifierProperty()` checks implicit modifiers as well.

Reply:
Ah, that makes sense. I didn't check if implicit modifiers are included and the docs weren't clear about it, but seeing the "workaround" I thought this needs to be handled manually.

I also added another method to the test so interface methods without explicit `public` modifer are actually covered too.